test/e2e/dr/common.go

```diff
@@ -0,0 +1,473 @@
+	var lastErr error
+	// gate on all clusteroperators being ready
+	available := make(map[string]struct{})
+	lastErr = nil
+	var lastCOs []objx.Map
+	wait.PollImmediate(30*time.Second, operatorWait, func() (bool, error) {
+		obj, err := coc.List(metav1.ListOptions{})
+		if err != nil {
+			lastErr = err
+			e2e.Logf("Unable to check for cluster operators: %v", err)
+			return false, nil
+		}
+		cv := objx.Map(obj.UnstructuredContent())
+		lastErr = nil
+		items := objects(cv.Get("items"))
+		lastCOs = items
+
+		if len(items) == 0 {
+			return false, nil
+		}
+
+		var unavailable []objx.Map
+		var unavailableNames []string
+		for _, co := range items {
+			if condition(co, "Available").Get("status").String() != "True" {
+				ns := co.Get("metadata.namespace").String()
+				name := co.Get("metadata.name").String()
+				unavailableNames = append(unavailableNames, fmt.Sprintf("%s/%s", ns, name))
+				unavailable = append(unavailable, co)
+				break
+			}
+			if condition(co, "Progressing").Get("status").String() != "False" {
+				ns := co.Get("metadata.namespace").String()
+				name := co.Get("metadata.name").String()
+				unavailableNames = append(unavailableNames, fmt.Sprintf("%s/%s", ns, name))
+				unavailable = append(unavailable, co)
+				break
+			}
+			if condition(co, "Failing").Get("status").String() != "False" {
+				ns := co.Get("metadata.namespace").String()
+				name := co.Get("metadata.name").String()
+				unavailableNames = append(unavailableNames, fmt.Sprintf("%s/%s", ns, name))
+				unavailable = append(unavailable, co)
+				break
+			}
+		}
+		if len(unavailable) > 0 {
+			e2e.Logf("Operators still doing work: %s", strings.Join(unavailableNames, ", "))
+			return false, nil
+		}
+		return true, nil
+	})
+
+	o.Expect(lastErr).NotTo(o.HaveOccurred())
+	var unavailable []string
+	buf := &bytes.Buffer{}
+	w := tabwriter.NewWriter(buf, 0, 4, 1, ' ', 0)
+	fmt.Fprintf(w, "NAMESPACE\tNAME\tPROGRESSING\tAVAILABLE\tVERSION\tMESSAGE\n")
+	for _, co := range lastCOs {
+		ns := co.Get("metadata.namespace").String()
+		name := co.Get("metadata.name").String()
+		if condition(co, "Available").Get("status").String() != "True" {
+			unavailable = append(unavailable, fmt.Sprintf("%s/%s", ns, name))
+		} else {
+			available[fmt.Sprintf("%s/%s", ns, name)] = struct{}{}
+		}
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n",
+			ns,
+			name,
+			condition(co, "Progressing").Get("status").String(),
+			condition(co, "Available").Get("status").String(),
+			co.Get("status.version").String(),
+			condition(co, "Failing").Get("message").String(),
+		)
+	}
+	w.Flush()
+	e2e.Logf("ClusterOperators:\n%s", buf.String())
+	if len(unavailable) > 0 {
+		e2e.Failf("Some cluster operators never became available %s", strings.Join(unavailable, ", "))
+	}
+	// Check at least one core operator is available
+	if len(available) == 0 {
+		e2e.Failf("There must be at least one cluster operator")
+	}
+}
+
+func restartSDNPods(oc *exutil.CLI) {
```

Comment:
I haven't found usage yet, but this stands out as odd.  Requiring this for a recovery seem like a bug.

Reply:
Its used in [quorum restore](https://github.com/openshift/origin/blob/5ba781f06d91836ffd068ba8ea0500c4237ff123/test/e2e/dr/quorum-restore.go#L241) - SDN needs to be restarted if the watches could not be restored

https://bugzilla.redhat.com/show_bug.cgi?id=1707006